src/main/java/engine/movegen/MovesGenerator.java

```diff
@@ -10,41 +11,114 @@
      */
     public boolean isOccupiedByFriendly (Move move, Board board) {
-        return true;
+        //ensure destination tile is occupied
+        if(board.getPieceAtAbsCoords(move.destination[0], move.destination[1]) != null){
+            Piece movingPiece = board.getPieceAtAbsCoords(move.origin[0], move.origin[1]);
+            Piece destPiece = board.getPieceAtAbsCoords(move.destination[0], move.destination[1]);
+            //returns true if moving and destination pieces have same colours
+            return(movingPiece.getColor() == destPiece.getColor());
+        }
+        //destination tile is not occupied
+        return false;
     }
-
     /**
-     * Checks to see whether destination of move is already
-     * occupied by an enemy piece using absolute coordinates.
+     * Checks to see whether destination of move is
+     * occupied by an enemy piece.
+     *
+     * @param move the move whose destination tile is being checked
+     * @param board the board on which the move exists
+     * @return whether the destination tile is occupied by an enemy piece
      */
     public boolean isOccupiedByEnemy (Move move, Board board) {
-        return true;
+        //ensure destination tile is occupied
+        if(board.getPieceAtAbsCoords(move.destination[0], move.destination[1]) != null){
+            Piece movingPiece = board.getPieceAtAbsCoords(move.origin[0], move.origin[1]);
+            Piece destPiece = board.getPieceAtAbsCoords(move.destination[0], move.destination[1]);
+            //returns true if moving and destination pieces have different colours
+            return(movingPiece.getColor() != destPiece.getColor());
+        }
+        //destination tile is not occupied
+        return false;
+    }
+
+    /**
+     * Generates moves in straight lines from an original tile in given directions. Includes captures.
+     * Does not include moves which land on friendly pieces, jump over pieces, or are off the board.
+     *
+     * @param origin the original coordinates of the piece for which moves are being generated
+     * @param directions the directions of moves to be generated, represented by coordinate changes
+     * @param board the board on which moves are being generated
+     * @return a list of valid straight line moves in the specified directions
+     */
+    public List<Move> genStraightLineMoves (int[] origin, int[][] directions, Board board){
+        List<Move> moveList = new ArrayList<Move>();
+        for(int[] direction: directions){
+            //loop through new tiles (up to the board's dimensions, i.e. 7) one tile at a time.
+            //the first tile which cannot be moved onto indicates that no further
+            //tile can be moved to, since it will either be jumping over pieces
+            //or off the board. Hence, terminate the loop when this happens.
+            for(int distance = 1; distance < Board.WIDTH; distance++){
```

Comment:
A while loop seems more natural. It might help avoid use of `break;` so that it is easier to trace code